# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/tests/browser/create_webrtc_peer_connection.html
# Commit: 812ed5275f0d
# Full Hash: 812ed5275f0db9f92944955f891df2ab68c9f8e7
# Author: Toshihito Kikuchi <tkikuchi@mozilla.com>
# Date: 2021-11-19 04:11:43
# Regressor Bug: 1734099
# File Overlap Count: 5
# Description:
#   Bug 1734099 - Never unload tabs with active RTCPeerConnection instances.  r=peterv,jesup
#   
#   This patch makes sure the Tab Unloading feature does not unload tabs that have
#   active peer connections not to disrupt browsing experience based on WebRTC peer
#   connections.
# ==============================================================================

diff -r a733d049dbda -r 812ed5275f0d dom/tests/browser/create_webrtc_peer_connection.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/tests/browser/create_webrtc_peer_connection.html	Thu Nov 18 22:59:12 2021 +0000
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+<meta charset="utf-8">
+<script>
+const gConnections = [];
+
+window.addEventListener("message", event => {
+  const ackTarget = window.parent ? window.parent : window;
+  switch (event.data) {
+    case "push-peer-connection":
+      gConnections.push(new RTCPeerConnection());
+      ackTarget.postMessage("ack", "*");
+      break;
+    case "pop-peer-connection":
+      gConnections.pop().close();
+      ackTarget.postMessage("ack", "*");
+      break;
+  }
+});
+
+window.addEventListener("DOMContentLoaded", function(ev) {
+  document.getElementById("msg").innerText = location.host;
+});
+</script>
+</head>
+<body><div id="msg"></div></body>
+</html>
